features/s3/objects.feature

```diff
@@ -215,3 +215,63 @@ Feature: Working with Objects in S3
   Scenario: Error handling
     Given I put "data" to the invalid key ""
     Then the status code should be 400
+
+  @bucket-slashes
+  Scenario: Sigv4 Bucket with trailing slash
+    Given I use signatureVersion "v4"
+    When I put "" to the key "" with bucket suffix "/"
+    Then the object "/" should exist
+  
+  @bucket-slashes
+  Scenario: Sigv4 Bucket with dobule trailing slashes
+    Given I use signatureVersion "v4"
+    When I put "" to the key "" with bucket suffix "//"
+    Then the object "//" should exist
+
+  @bucket-slashes
+  Scenario: Sigv4 Bucket with slashes without trailing slash
+    Given I use signatureVersion "v4"
+    When I put "" to the key "" with bucket suffix "/a/b"
+    Then the object "a/b/" should exist
+
+  @bucket-slashes
+  Scenario: Sigv4 Bucket with slashes with Key
+    Given I use signatureVersion "v4"
+    When I put "" to the key "foo" with bucket suffix "/a/b"
+    Then the object "a/b/foo" should exist
+
+  @bucket-slashes
+  Scenario: Sigv4 Bucket with trailing slashes with Key
+    Given I use signatureVersion "v4"
+    When I put "" to the key "foo" with bucket suffix "/a/b/"
+    Then the object "a/b//foo" should exist
+
+  @bucket-slashes
+  Scenario: Sigv2 Bucket with trailing slash
+    Given I use signatureVersion "v2"
+    When I put "" to the key "" with bucket suffix "/"
+    Then the object "/" should exist
+  
+  @bucket-slashes
+  Scenario: Sigv2 Bucket with double trailing slashes
+    Given I use signatureVersion "v2"
+    When I put "" to the key "" with bucket suffix "//"
+    Then the object "//" should exist
+
+  @bucket-slashes
+  Scenario: Sigv2 Bucket with slashes without trailing slash
+    Given I use signatureVersion "v2"
+    When I put "" to the key "" with bucket suffix "/a/b"
+    Then the object "a/b/" should exist
+
+  @bucket-slashes
+  Scenario: Sigv2 Bucket with slashes with Key
+    Given I use signatureVersion "v2"
+    When I put "" to the key "foo" with bucket suffix "/a/b"
+    Then the object "a/b/foo" should exist
+
+  @bucket-slashes
+  Scenario: Sigv2 Bucket with trailing slashes with Key
+    Given I use signatureVersion "v2"
+    When I put "" to the key "foo" with bucket suffix "/a/b/"
+    Then the object "a/b//foo" should exist
```

Comment:
We can't change it, but this behavior does not seem completely intuitive. Bash at least will treat one or more slashes in a sequence as a single directory separator. Just something to keep in mind for any future high-level interfaces, especially those that allow/encourage users to interact with S3 as if it were a filesystem.